parlai/mturk/tasks/dealnodeal/worlds.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+from parlai.mturk.core.worlds import MTurkTaskWorld
+from parlai.core.worlds import validate
+from joblib import Parallel, delayed
+from parlai.tasks.dealnodeal.agents import NegotiationTeacher
+import random
+
+class MTurkDealNoDealDialogWorld(MTurkTaskWorld):
+    """World where two agents have a dialogue to negotiate a deal.
+    """
+    def __init__(self, opt, agents=None, shared=None):
+        # Add passed in agents directly.
+        if agents is not None:
+            random.shuffle(agents)
+        self.agents = agents
+        self.acts = [None] * len(agents)
+        self.episodeDone = False
+        self.task = NegotiationTeacher(opt=opt)
+        self.first_turn = True
+        self.choices = dict()
+        self.selection = False
+        self.turns = 0
+        self.num_negotiations = 0
+
+
+    def parley(self):
+        """Alternate taking turns, until both agents have made a choice (indicated by a turn starting with <selection>)
+        """
+        if self.first_turn:
+            self.num_negotiations += 1
+
+            # Use NegotiationTeacher to load data for us
+            self.task.dialogue_idx = None
+            act = self.task.act()
+            for other_agent in self.agents:
+                other_agent.observe(validate(act))
```

Comment:
It would appear that the NegotiationTeacher is giving both of the agents the same values for both items here. Shouldn't there be something that provides different values to each of the agents while maintaining the same counts?